### PR TITLE
[desktop_drop]: Fixes false assertions raised in web

### PR DESCRIPTION
**Pull request by paschalisp**
* Removed unneeded asserts in _handleMethodChannel (_offset == null) as variable's nullability is handled anyways in the code (was causing wrong assertions in web, perhaps they're forgotten assertions?)

* Updated cross_file to package's latest version

* Tested for web and linux target platforms

### CODE REVIEW

File: packages/desktop_drop/lib/src/channel.dart [hunk 1]
```diff
@@ -40,7 +40,6 @@ class DesktopDrop {
   Future<void> _handleMethodChannel(MethodCall call) async {
     switch (call.method) {
       case "entered":
-        assert(_offset == null);
```
Comment (boyan01) [inline packages/desktop_drop/lib/src/channel.dart:43]: The assertion here is to guarantee that:
1. The `entered` event will not be received repeatedly.
2. It is no longer in the dropping state, that is, it has received an `exit` or `performOperation` event.

File: packages/desktop_drop/lib/src/channel.dart [hunk 2]
```diff
@@ -75,10 +71,8 @@ class DesktopDrop {
         break;
       case "performOperation_linux":
         // gtk notify 'exit' before 'performOperation'.
-        assert(_offset == null);
```
Comment (boyan01) [inline packages/desktop_drop/lib/src/channel.dart:78]: The assertion here is to guarantee that:

When the `performOperation` event is received, it is currently in the dropping state, that is, it has received the `entered` event before

File: packages/desktop_drop/lib/src/channel.dart [hunk 3]
```diff
@@ -93,7 +87,6 @@ class DesktopDrop {
         ));
         break;
       case "performOperation_web":
-        assert(_offset != null);
```
Comment (boyan01) [inline packages/desktop_drop/lib/src/channel.dart:96]: The assertion here is to guarantee that:

When the `performOperation` event is received, it is currently in the dropping state, that is, it has received the `entered` event before

**Comment by boyan01** [comment]
Thanks for your PR.

According the comments before, in my thought, the assertions in `_handleMethodChannel` are needed.

Of course, in fact, these assertions will not affect the operation of profile / release, they will only be valid in debug mode.

And if there is a wrong assertions in web, Instead of simply removing it, we need to figure out under what circumstances this assertion would cause an error.

**Comment by paschalisp** [comment]
Thanks for your reply and *special thanks* for your great library. In our case the assertions were causing exceptions during debugging making the development of the specific views impossible, especially their uploading functionality.

I'm sure they were there for a reason, but perhaps with the latest framework/dependent package versions you might need to re-examine in which conditions the _offset shall be null and in which shall have a value?

**Comment by rahmanrezaee** [comment]
thanks for your library 
please merge the request

**Comment by walsha2** [comment]
@boyan01 This is related to #103 and that issue is very old, but might be worth another look today, a year later. 

To be clear though, even though the assertion error happens, it seems that the drag event is still registered (eventually) so unless you are looking at the logs you might not notice that there is anything wrong. The drag/drop works... although with many assert errors emitted.

On web, these `_offset == null` are repeated many many times when the drag area is entered/exited, but the ultimate event seems to register. It looks like _some_ events cause this error and some do not. Can we get a revisit of #103 and this assert logic specifically? I agree with @paschalisp comments:

> I'm sure they were there for a reason, but perhaps with the latest framework/dependent package versions you might need to **re-examine in which conditions** the _offset shall be null and in which shall have a value?

**Comment by boyan01** [comment]
Sorry for later. 

If these asserts really affect development, removing them would be acceptable to me.

SGTM

**Pull request merged**